src/provider/mod.rs

```diff
@@ -24,28 +24,47 @@ where
     T: Transport + Clone,
 {
     /// Gets the address of the main ZKsync contract on L1.
+    /// # Returns
```

Comment:
Nit: the common approach (at least one that I've seen) was to have an empty line before header and use `##` instead of `#`.

Although, given that here and below the header is mostly identical to the `Returns` section -- do we need it? Do you think it improves accessibility (genuine question)?

Reply:
@popzxc I don't think that _Returns_ with mostly identical text is helpful. I added them for consistency, but as I think about it more, I'm leaning towards deleting the non-informative ones. Wdyt about parameters annotations that are also obvious, e.g. `tx_hash` - hash of the transaction.

Reply:
I think that if there is at least one non-obvious parameter, it's OK to describe the obvious ones as well. If all the parameters are obvious, the section can be omitted.

Reply:
Applied the nit and removed non-informative annotations.